exporters/etw/include/opentelemetry/exporters/etw/etw_provider_exporter.h

```diff
@@ -0,0 +1,657 @@
+        else
+        {
+          data.providerHandle = hProvider;
+        };
+      };
+      break;
+#endif
+
+#ifdef HAVE_MSGPACK
+      // Register for MsgPack payload ETW events.
+      case EventFormat::ETW_MSGPACK:
+      {
+        REGHANDLE hProvider = 0;
+        if (EventRegister(&data.providerGuid, NULL, NULL, &hProvider) != ERROR_SUCCESS)
+        {
+          // There was an error registering the ETW provider
+          data.providerHandle = INVALID_HANDLE;
+        }
+        else
+        {
+          data.providerHandle = hProvider;
+        }
+      };
+      break;
+#endif
+
+      default:
+        // TODO: other protocols, e.g. XML events - not supported yet
+        break;
+    }
+
+    // We always return an entry even if we failed to register.
+    // Caller should check whether the hProvider handle is valid.
+    return data;
+  }
+
+  /// <summary>
+  /// Unregister Provider
+  /// </summary>
+  /// <param name="providerId"></param>
+  /// <returns></returns>
+  unsigned long close(Handle data)
+  {
+    std::lock_guard<std::mutex> lock(m_providerMapLock);
+
+    auto m  = providers();
+    auto it = m.begin();
+    while (it != m.end())
+    {
+      if (it->second.providerHandle == data.providerHandle)
+      {
+        auto result = EventUnregister(data.providerHandle);
+        m.erase(it);
+        return result;
+      }
+    };
+    return STATUS_ERROR;
+  }
+
+#ifdef HAVE_MSGPACK
+  template <class T>
+  unsigned long writeMsgPack(Handle &providerData, T eventData)
+  {
+
+    // Make sure you stop sending event before register unregistering providerData
+    if (providerData.providerHandle == INVALID_HANDLE)
+    {
+      // Provider not registered!
+      return STATUS_ERROR;
+    };
+
+    const std::string EVENT_NAME = "name";
+    std::string eventName        = "NoName";
+    auto nameField               = eventData[EVENT_NAME];
+    switch (nameField.index())
+    {
+      case common::AttributeType::TYPE_STRING:
+        eventName =
+            (char *)(nostd::get<nostd::string_view>(nameField).data());  // must be 0-terminated!
+        break;
+      case common::AttributeType::TYPE_CSTRING:
+        eventName = (char *)(nostd::get<const char *>(nameField));
+        break;
+      default:
+        // This is user error. Invalid event name!
+        // We supply default 'NoName' event name in this case.
+        break;
+    }
+
+    /* clang-format off */
+    nlohmann::json jObj =
+    {
+      { "env_name", "Span" },
+      { "env_ver", "4.0" },
+      //
+      // TODO: compute time in MessagePack-friendly format
+      // TODO: should we consider uint64_t format with Unix timestamps for ELK stack?
+      { "env_time",
+        {
+          { "TypeCode", 255 },
+          { "Body","0xFFFFFC60000000005F752C2C" }
+        }
+      },
+      //
+
+      // TODO: follow JSON implementation of OTLP or place protobuf for non-Microsoft flows
+      { "env_dt_traceId", "6dcdae7b9b0c7643967d74ee54056178" },
```

Comment:
@mishal23 - please log a separate issue to properly populate this. Currently it's hardcoded at some bogus value, but it has to be captured from the originating span data.

Reply:
Will be addressed in #466